src/game/objects/traps/movable_block.c

```diff
@@ -133,13 +139,19 @@ void MovableBlock_Collision(
 {
     ITEM_INFO *item = &g_Items[item_num];
 
+    if (Item_TestAnimEqual(lara_item, LA_STOP)
+        || g_Lara.water_status == LWS_CHEAT) {
```

Comment:
This still causes a problem.

https://youtu.be/fmq2PW45mxE

WDYT about just doing this in the condition?

```c
if (item->current_anim_state == MBS_STILL) {
```

This allows the block's animation to complete if Lara enters DOZY. There are only three states for the block: still (covers falling too), being pushed, being pulled. And adding to this, maybe we could remove `m_BlockPushPull` and just look at the state instead?

Reply:
Yes I think that works. I'll push the change.